server_test.go

```diff
@@ -677,3 +677,36 @@ zDCJkckCgYEAndqM5KXGk5xYo+MAA1paZcbTUXwaWwjLU+XSRSSoyBEi5xMtfvUb
 kFsxKCqxAnBVGEWAvVZAiiTOxleQFjz5RnL0BQp9Lg2cQe+dvuUmIAA=
 -----END RSA PRIVATE KEY-----`)
 )
+
+func testShutdownBindPort(t *testing.T, protocol string, port string) {
+	handler := NewServeMux()
+	handler.HandleFunc(".", func(w ResponseWriter, r *Msg) {})
+	s := &Server{
+		Addr:    net.JoinHostPort("127.0.0.1", port),
+		Net:     protocol,
+		Handler: handler,
+	}
+	go func() {
+		if err := s.ListenAndServe(); err != nil {
+			t.Log(err)
+		}
+	}()
+	time.Sleep(500 * time.Millisecond)
+	if err := s.Shutdown(); err != nil {
+		t.Fatal(err)
+	}
```

Comment:
@miekg if I add a small time.Sleep duration here, the test will never fail. But if I don't sometimes it fails. I guess the linux network stack still needs a small amount time after closing the connection before  it can be used again. I might depends on some kernel parameters but I can't figure out exactly.

So what can we do about it?

- adds a small sleep to Shutdown function if os is linux
- accepts it and documents somewhere .